Core/Src/dashboard.c

```diff
@@ -388,63 +382,20 @@ void SetupDashBoard(void) {
-    PCA9555_pinMode(&pca9555Handle, 9, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-    PCA9555_pinMode(&pca9555Handle, 10, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-    PCA9555_pinMode(&pca9555Handle, 11, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-    PCA9555_pinMode(&pca9555Handle, 12, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-    PCA9555_pinMode(&pca9555Handle, 13, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-    PCA9555_pinMode(&pca9555Handle, 14, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-    PCA9555_pinMode(&pca9555Handle, 15, PCA9555_PIN_OUTPUT_MODE, PCA9555_POLARITY_NORMAL);
-
-    PCA9555_digitalWrite(&pca9555Handle, 0, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 1, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 2, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 3, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 4, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 5, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 6, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 7, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 8, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 9, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 10, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 11, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 12, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 13, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 14, PCA9555_BIT_RESET);
-    PCA9555_digitalWrite(&pca9555Handle, 15, PCA9555_BIT_RESET);
-
-    HAL_GPIO_WritePin(NHD_C0220BIZx_nRST_GPIO_OUT_GPIO_Port, NHD_C0220BIZx_nRST_GPIO_OUT_Pin, GPIO_PIN_SET);
-
-    LCD_DisplayHandle.LCD_hi2c              = &hi2c1;
-    LCD_DisplayHandle.LCD_htim_backlight    = NULL;
-    LCD_DisplayHandle.TIM_channel_backlight = 0;
-    LCD_DisplayHandle.i2cAddr               = 0x78;
-    LCD_DisplayHandle.num_col               = 20;
-    LCD_DisplayHandle.num_lines             = 2;
-
-    LCD_ST7032_Init(&LCD_DisplayHandle);  // Init LCD
-    LCD_clear();                          // clear LCD
-    LCD_home();                           // Home The display
-    LCD_contrast(15);                     // set contrast to level 15 - MAXIMUM (15 level available)
-
-    char msg[54] = {0};
-    sprintf(msg, "Dashboard 2022 Boot - build %s @ %s\r\n", __DATE__, __TIME__);
-    HAL_UART_Transmit(&huart1, (uint8_t *)msg, strlen(msg), 20);
+    ILI9488_init();
+    lv_init();
+
+    lv_tick_set_cb(HAL_GetTick);
+    lv_display_t *display1 = lv_display_create(HORIZONTAL_RES, VERTICAL_RES);
+    lv_display_set_buffers(display1, buf1, NULL, BUFFER_SIZE, LV_DISPLAY_RENDER_MODE_PARTIAL);
+    lv_display_set_flush_cb(display1, LVGL_CLB_flush_clb);
+
+#if LV_USE_LOG
+    lv_log_register_print_cb(LVGL_CLB_log_clb);
+#endif
```

Comment:
merge in an autoesplicative function, e.g.: `GUI_init()`. You could have a `gui.c/h` module somewhere where you put all this middle functions.